### ISSUE TITLE
additional onClick event when enableMouseEvents: true

### ISSUE THREAD

**Issue by Ognian** [yahoo/react-dnd-touch-backend#13]
I'm using this backend with enableMouseEvents: true.
It works great on the iPad but on the Desktop I have the following problem:
I'm dragging Buttons around, so that they can be rearranged and clicked, but when such a button is dragged around and dropped somewhere I get also a onClick event, without actually being clicked (just dropped). Looks like some preventDefault() is missed?
Thanks
Ognian

**Comment by longlho** [comment]
mouse events are inherently troublesome due to unintentional propagation and different element behaviors (links, buttons, images behave differently). I'd recommend using [HTML5 dnd backend](https://github.com/gaearon/react-dnd-html5-backend) if you don't have to support hybrid devices. What's your use case?

**Comment by Ognian** [comment]
I'm supporting touch devices (actually iPad) and Desktop (with mouse only and no touch), so of course I can try to distinguish on wich device I'm running and use a different backend. I just tougth that this is a "small" problem ...

**Comment by longlho** [comment]
You're right that it's easy to slap a `preventDefault`/`stopPropagation` in there, but we've had reports where people want events to propagate properly, so it's really a preference.

**Comment by kmkr** [comment]
Did you figure out how to solve this problem, @Ognian ?

**Comment by Ognian** [comment]
For the moment I'm simply ignoring this problem :-) It should be possible to detect if it is a desktop browser and then use the HTML5 and backend instead, but I haven't tried to do so.

**Comment by kmkr** [comment]
Yes I was trying to, but the closest I came was to check user agent headers. If there was a way to either detect touchevent-support OR native drag support that would be better. It seems like some browsers state they have native drag support even though the don't, and it's a similar problem with touch screens.

I tried adding `.preventDefault()` and `.stopPropagation()` in a fork of this project, but I couldn't get it to work. Do you know which of the handlers are propagating the click event?

**Comment by Ognian** [comment]
I haven't dig in deeper, so no idea how to fix it. In my case I have either a Desktop Browser with no touch or an iPad, so I would check for the IPad and use the Touch handler in this case and in all other cases the one without touch support.

**Comment by jrodl3r** [comment]
@longlho where would the `preventDefault/stopPropagation` go in the package to stop the `onClick` from firing on parent elements? Thx!

**Comment by longlho** [comment]
if u `preventDefault` on any of the touch events, mouse simulation won't be fired
